tower/src/util/boxed/layer.rs

```diff
@@ -0,0 +1,86 @@
+use crate::util::BoxService;
+use std::{fmt, sync::Arc};
+use tower_layer::{layer_fn, Layer};
+use tower_service::Service;
+
+/// A boxed `Layer` trait object.
```

Comment:
nit: should `Layer` be a link?
```suggestion
/// A boxed [`Layer`] trait object.
```